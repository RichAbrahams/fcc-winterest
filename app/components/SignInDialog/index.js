/**
*
* SignInDialog
*
*/

import React from 'react';
import SignInDialogueWrapper from 'components/SignInDialogueWrapper';
import SignInDialogueTitle from 'components/SignInDialogueTitle';
import Button from 'components/SignInDialogueButton';
import FA from 'components/FontAwesome';


function SignInDialog({ signIn }) {
  return (
    <SignInDialogueWrapper>
      <SignInDialogueTitle>Sign In</SignInDialogueTitle>
      <Button signIn={signIn} service="google">
        <FA
          name="google"
          size="2x"
          color="white"
        />
      </Button>
      <Button signIn={signIn} service="github">
        <FA
          name="github"
          size="2x"
          color="white"
        />
      </Button>
      <Button signIn={signIn} service="twitter">
        <FA
          name="twitter"
          size="2x"
          color="white"
        /></Button>
    </SignInDialogueWrapper>
  );
}

SignInDialog.propTypes = {
  signIn: React.PropTypes.func,
};

export default SignInDialog;
