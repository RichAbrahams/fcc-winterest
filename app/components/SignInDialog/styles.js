import styled from 'styled-components';


export const Button = styled.button `
  width: 10em;
  padding: 1em;
  color: #444;
  border: 3px solid #444;
  margin: 0.5em 1em 0em 1em;
  &:hover {
    background: purple;
  }
`;

