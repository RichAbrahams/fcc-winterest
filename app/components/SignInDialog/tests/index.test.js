import React from 'react';
import renderer from 'react-test-renderer';
import SignInDialog from '../index';

it('renders correctly', () => {
  const tree = renderer.create(
    <SignInDialog />).toJSON();
  expect(tree).toMatchSnapshot();
});
