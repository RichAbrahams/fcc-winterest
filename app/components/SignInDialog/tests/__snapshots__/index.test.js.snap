exports[`test renders correctly 1`] = `
<div
  className="eMxMZS"
  onClick={[Function]}>
  <h2
    className="cTfNJF">
    Sign In
  </h2>
  <button
    className="jgFpNU"
    onClick={[Function]}>
    <span
      aria-hidden={true}
      className="fa fa-google fa-2x jYQlWx"
      color="white"
      transform={undefined} />
  </button>
  <button
    className="jgFpNU"
    onClick={[Function]}>
    <span
      aria-hidden={true}
      className="fa fa-github fa-2x jYQlWx"
      color="white"
      transform={undefined} />
  </button>
  <button
    className="jgFpNU"
    onClick={[Function]}>
    <span
      aria-hidden={true}
      className="fa fa-twitter fa-2x jYQlWx"
      color="white"
      transform={undefined} />
  </button>
</div>
`;
